cirq-core/cirq/testing/repr_pretty_tester.py

```diff
@@ -53,3 +53,22 @@ def assert_repr_pretty(val: Any, text: str, cycle: bool = False):
     p = FakePrinter()
     val._repr_pretty_(p, cycle=cycle)
     assert p.text_pretty == text, f"{p.text_pretty} != {text}"
+
+
+def assert_repr_pretty_contains(val: Any, substr: str, cycle: bool = False):
```

Comment:
Might be nice to add a version that does a more general regex match as well:
```python
def assert_repr_pretty_matches(val: Any, pattern: Union[str, re.Pattern], cycle: bool = False):
    p = FakePrinter()
    val._repr_pretty(p, cycle=cycle)
    assert re.match(pattern, p.text_pretty), ...
```

Reply:
I appeal to YAGNI here - if someone needs it, they'll make it.